### ISSUE TITLE
How to refresh a row on button click

### ISSUE THREAD

**Issue by Jeru2023** [gradio-app/gradio#7498]
### Describe the bug

in below code i want to load a complex multitab dataframe in result_row:
```
        with gr.Row():
            with gr.Column():
                refresh_btn = gr.Button(value="Refresh", variant="primary")

                gr.Markdown("## Evaluation Results")

                result_row = gr.Row()
                refresh_btn.click(refresh, datasets, result_row)
```

I’ve defined a refresh function to render result row
```
def refresh(datasets):
    with gr.Row() as result_row:
        for dataset in datasets:
            print('**** ', dataset)
            with gr.Tab(dataset):
		# just placeholder, i will replace it with dataframe later
                gr.TextArea(label="Evaluation Result", interactive=False, lines=30)
    return result_row
```

After I click refresh button nothing happend, nothing change in the ui, no error either.

### Logs

****  T2Retrieval
****  MMarcoRetrieval
****  EcomRetrieval

### System Info

```shell
Gradio Environment Information:
------------------------------
Operating System: Darwin
gradio version: 4.19.1
gradio_client version: 0.10.0
```

### Severity

I can work around it

**Comment by freddyaboulton** [comment]
Hi @Jeru2023 ! In gradio you cannot dynamically create components from an event handler. In this case, I think you should return an update for each dataframe, e.g. `return gr.DataFrame(...), gr.DataFrame(...)`

Linking this guide for reference just in case: https://www.gradio.app/guides/blocks-and-event-listeners. Going to close as this is not a gradio bug

**Comment by Jeru2023** [comment]
Thanks Freddy, I made it work by refresh the entire data frame, it's a redundant workaround.
I'm wondering if Gradio will support async callback in the future, that will make this kind of requirement much easier.

**Comment by Jeru2023** [comment]
It's just like the old concept in html and javascript ajax.

I defined a stab or placeholder container A with an element id a001.
after I clicked a button, I can dynamically rerender this container with whatever complex layout or nested components.

In gradio i feel like there is a strict restriction on dynamic repaint. The only solution is we define some components with false visibility and then replace them with real output, that will produce a lot of code redundancy.
